src/webview.cpp

```diff
@@ -49,6 +49,25 @@ WebView::WebView(QWidget *parent)
     connect(this->page(), &QWebEnginePage::linkHovered, this, [=] (const QString& url) {
         m_linkHovered = url;
     });
+
+    /* This rezooms the page to its correct zoom(default/by ZIM ID) after loading is finished.
+    If the page is result of fulltext search, use the default zoom
+    Needed because of https://bugreports.qt.io/browse/QTBUG-51851 */
+    connect(this, &QWebEngineView::loadFinished, this, [=] (bool ok) {
+        if (ok) {
+            auto url = this->url();
+            auto settingsManager = KiwixApp::instance()->getSettingsManager();
+            qreal zoomFactor;
+            QUrlQuery query(url);
+            if (query.hasQueryItem("pattern")) {
+               zoomFactor = settingsManager->getZoomFactor();
+            } else {
+                auto zimId = url.host().split('.')[0];
```

Comment:
I feel suspicious about a ZIM id obtained in such a way. I see that it is a copy from `WebView::onUrlChanged()`. I think that we have a bug there. For example, the URL `http://library.kiwix.org/phet_hy_2021-08` will have `library` extracted from it as `zimId` whereas my impression was that we are interested in the `phet_hy_2021-08` part.

@kelson42 Will you please confirm that my understanding is correct?

Reply:
On kiwix-desktop, the url format is `zimid.zim/article/path`.
This is somehow a stable fact as the custom url handler is installed to respond to all `zim` tld (Other request are not catch, and so we can transfer to native browser for external url)
Extracting the zimId from the url could (should) be move to a dedicated function, but it is a correct way to obtain the zimId (In the customSchemeHandler, we simply remove the 4 last chars)